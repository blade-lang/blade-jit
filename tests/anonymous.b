@{
  echo 'Self called'
}()

var fn = @{
  return 10
}

var fn2 = @{
  return 20
}

var fn3 = @(val) {
  return val ** 5
}

var fn4 = @(val, ...arg) {
  return '${val} is not in list ${arg}'
}

echo fn()
echo fn2()
echo fn3(14)
echo fn4(40, 50, 60)

class Demo {
  var multiply = @(val) {
    self.value = 1_000 * val
    return val * 30
  }

  get_val() {
    return self.value
  }
}

const demo = new Demo()
echo demo.multiply(15)
echo demo.get_val()
echo Demo.multiply(46)
