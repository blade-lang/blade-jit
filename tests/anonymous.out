Self called
10
20
537824
40 is not in list [50, 60]
450
15000
1380